src/wp-includes/theme.php

```diff
@@ -840,6 +840,16 @@ function switch_theme( $stylesheet ) {
 	$new_theme->delete_pattern_cache();
 	$old_theme->delete_pattern_cache();
 
+	// Set autoload=no for the old theme, autoload=yes for the switched theme.
+	$theme_mods_options = array(
+		'theme_mods_' . $stylesheet                  => 'yes',
+		'theme_mods_' . $old_theme->get_stylesheet() => 'no',
+	);
```

Comment:
What about the respective `template`? For both the old and new theme, we'll also have to add their respective template's `theme_mods_*` entries here _if_ their `template` differs from the respective `stylesheet`.

_And_ last but not least (edge-case, but certainly can happen): We need to ensure that if both themes are child themes of the same parent theme, that the `template` remains at / is set to `yes`, i.e. there must be only one entry, and the `yes` takes precedence over `no`.

Reply:
Does a parent theme's theme mods get merged with a child them when active? I thought only the active theme's options got used at any one time.

Reply:
Oof. You're right.

I guess that's not great for data portability between a parent and child theme, but that's a completely separate topic. The way it works now the code here looks good then.